src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import { Loader } from '../Loader';
+import { useFetch } from '../hooks/useFetch';
+import { User } from '../../types/User';
+import { useAppSelector } from '../../app/store';
+import { getUser } from '../../api';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoModal: React.FC = () => {
+  const todo = useAppSelector(state => state.currentTodo) as Todo;
+  const dispatch = useDispatch();
+  const { setCurrentTodo } = currentTodoSlice.actions;
+  const fetchUser = useCallback(() => {
+    return todo ? getUser(todo.userId) : Promise.resolve({} as User);
+  }, [todo]);
+
+  const {
+    data: user,
+    errorMessage,
+    isLoading,
+  } = useFetch<User>(fetchUser, {} as User);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{todo?.id}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(setCurrentTodo(null))}
```

Comment:
TIP: Try minimizing the use of anonymous functions. You can describe the handler for this case.